classwork/classwork5.2/MegaDataPool.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "MegaData.h"
+#include <vector>
+#include <mutex>
+
+class MegaDataPool {
+public:
+    static MegaDataPool& instance(size_t poolSize = 10);
+    MegaData* acquire();
```

Comment:
it's definitely should be marked as nodiscard cause ignoring the result will mean that no one will release that pointer.
Also we can return & here intead of * cause * can be unintentionally passed to delete.